### PR TITLE
Bug Fix - Select an invite from the notifications doesn't not work

### PR DESCRIPTION
**Pull request by giomfo**
#275

### CODE REVIEW

File: Tchap/Modules/Application/AppCoordinatorType.swift [hunk 1]
```diff
@@ -27,6 +27,12 @@ protocol AppCoordinatorType: Coordinator {
     /// - Returns: true to indicate that the activity has been handled, or false to let iOS handle the activity.
     func handleUserActivity(_ userActivity: NSUserActivity, application: UIApplication) -> Bool
     
+    /// Resume the application by selecting a room.
+    ///
+    /// - Parameters:
+    ///   - roomId: the room identifier.
+    func resumeBySelectingRoom(with roomId: String)
```
Comment (SBiOSoftWhare) [inline Tchap/Modules/Application/AppCoordinatorType.swift]: Maybe you could simply add a method  `func  userDidTapNotification(with userInfo: [String: Any])` and handle the action to make in `AppCoordinator`

File: Tchap/Modules/Application/AppDelegate.swift [hunk 2]
```diff
@@ -109,7 +109,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             return
         }
         
-        _ = self.appCoordinator.showRoom(with: roomId)
+        _ = self.appCoordinator.resumeBySelectingRoom(with: roomId)
```
Comment (SBiOSoftWhare) [inline Tchap/Modules/Application/AppDelegate.swift]: Remove usage of deprecated methods `application(_:handleActionWithIdentifier:for:completionHandler:)` and `application(_:didReceive:)` and use `userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` to navigate to a room instead.

File: Tchap/Modules/Rooms/RoomsCoordinator.swift [hunk 3]
```diff
@@ -73,6 +73,16 @@ final class RoomsCoordinator: NSObject, RoomsCoordinatorType {
         self.roomsDataSource.search(withPatterns: pattern)
     }
     
+    func scrollToRoom(with roomID: String, animated: Bool) {
+        if let indexPath = self.roomsDataSource.cellIndexPath(withRoomId: roomID, andMatrixSession: self.session) {
```
Comment (SBiOSoftWhare) [inline Tchap/Modules/Rooms/RoomsCoordinator.swift]: Prefer `guard let` here

File: Tchap/Modules/Application/AppDelegate.swift (hunk 2 above)
Comment (giomfo) [reply Tchap/Modules/Application/AppDelegate.swift]: done during the rebase (develop_rebase branch)

File: Tchap/Modules/Rooms/RoomsCoordinator.swift (hunk 3 above)
Comment (giomfo) [reply Tchap/Modules/Rooms/RoomsCoordinator.swift]: done

**Pull request merged**